run/shas.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import logging
+import os
+import subprocess
+
+from datetime import date, timedelta
+
+"""
+shas.py defines methods for determining which SHA to run against on a given
+date.
+"""
+
+
+class SHAFinder(object):
+
+    def __init__(self,
+                 logger  # type: logging.Logger
+                 ):
+        self.logger = logger
+
+    def get_todays_sha(self, path):  # type: (str) -> str
+        """
+        Gets the first SHA for the current date of the git repo in the given
+        path.
+        """
+        today = date.today()
+        tomorrow = today + timedelta(days=1)
+        command = [
+            'git',
+            'log',
+            '--format=%H',
+            '--after="%s 00:00:00"' % today.isoformat(),
+            '--before="%s 00:00:00"' % tomorrow.isoformat(),
+            '--date-order',
```

Comment:
`--date-order` will make this less stable than it could be I think. Just dropping that should result first commit in tree order after the cutoff being used, even if there later appears a commit with an earlier timestamp. Might want to verify with tests though :)

Reply:
Test still passes.. so sure :)